spec/index.bs

```diff
@@ -2953,14 +2960,42 @@ interface GPUQueue {
 GPUQueue includes GPUObjectBase;
 </script>
 
- - {{GPUQueue/copyImageBitmapToTexture()}}:
-   - For now, {{GPUQueue/copyImageBitmapToTexture(source, destination, copySize)/copySize}}.[=Extent3D/depth=] must be `1`.
+<dl dfn-type="method" dfn-for="GPUQueue">
+    : <dfn>writeBuffer(sourceData, sourceOffset, destination, destinationOffset, size)</dfn>
+    ::
+        Takes the `sourceData` contents of size `size`, starting from the byte offset `sourceOffset`,
+        and schedules a write operation of these contents to the `destination` buffer on the
+        [=Queue timeline=] starting at `destinationOffset`.
+        Any subsequent modifications to `sourceData` do not affect what is written
+        at the time that the scheduled operation runs.
+
+        The operation does nothing and produces an error if any of the following is true:
+
+        - `destination` buffer wasn't created with {{GPUBufferUsage/COPY_DST}} flag in {{GPUBufferDescriptor/usage}}.
+        - `destination` buffer is destroyed
+        - `destinationOffset + size` exceeds the {{GPUBufferDescriptor/size}} of the `destination` buffer.
+        - `destination` buffer isn't in the `"unmapped"` [=buffer state=].
+        - `destinationOffset` is not a multiple of 4.
+        - `size` is not a positive multiple of 4.
+        - `sourceOffset + size` exceeds `sourceData.byteLength`.
```

Comment:
nit: I haven't been following spec reviews, so I may be out of date. Do we differentiate when an operation would throw an exception? I would think this would be an exception and not an asynchronous error because it's an out-of-bounds programming error that can be checked synchronously.

Reply:
You're right, I think this should be an exception. We should differentiate it now so we don't lose track. Can just write `throws {{OperationError}} if ...` and we'll figure out how to formalize it eventually.

Probably the buffer state check and alignment checks should also be exceptions.